mir_eval/chord.py

```diff
@@ -1342,6 +1347,134 @@ def sevenths_inv(reference_labels, estimated_labels):
     return comparison_scores
 
 
+def directional_hamming_distance(reference_intervals, estimated_intervals):
+    """Compute the directional hamming distance between reference and
+    estimated intervals as defined by [#harte2010towards]_ and used for MIREX
+    'OverSeg', 'UnderSeg' and 'MeanSeg' measures.
+
+    Examples
+    --------
+    >>> (ref_intervals,
+    ...  ref_labels) = mir_eval.io.load_labeled_intervals('ref.lab')
+    >>> (est_intervals,
+    ...  est_labels) = mir_eval.io.load_labeled_intervals('est.lab')
+    >>> overseg = 1 - mir_eval.chord.directional_hamming_distance(
+    ...     ref_intervals, est_intervals)
+    >>> underseg = 1 - mir_eval.chord.directional_hamming_distance(
+    ...     est_intervals, ref_intervals)
+    >>> meanseg = min(overseg, underseg)
+
+    Parameters
+    ----------
+    reference_intervals : np.ndarray, shape=(n, 2), dtype=float
+        Reference chord intervals to score against.
+    estimated_intervals : np.ndarray, shape=(m, 2), dtype=float
+        Estimated chord intervals to score against.
+
+    Returns
+    -------
+    directional hamming distance : float
+        directional hamming distance between reference intervals and
+        estimated intervals.
+    """
+    est_ts = np.unique(estimated_intervals.flatten())
```

Comment:
Seems like this should have an input validation

Reply:
Something like `if estimated_intervals.shape[1] != 2` or should the check go deeper and ensure e.g. that there are no overlaps in the intervals, etc.?

Reply:
Calling `util.validate_intervals` would probably suffice.  It doesn't check for disjoint/complete segmentation (and maybe it should), but it does cover the basics of shape matching and valid interval timings.